ledger-tool/src/main.rs

```diff
@@ -628,6 +629,18 @@ fn main() {
         .takes_value(true)
         .default_value(&default_genesis_archive_unpacked_size)
         .help("maximum total uncompressed size of unpacked genesis archive");
+    let snapshot_version_arg = Arg::with_name("snapshot_version")
+        .long("snapshot-version")
+        .value_name("VERSION")
+        .validator(|arg| {
+            arg.parse::<snapshot_utils::SnapshotVersion>()
+                .map(|_| ())
+                .map_err(|e| e.to_string())
+        })
+        .multiple(false)
+        .required(false)
+        .takes_value(true)
+        .help("Output snapshot in this version");
```

Comment:
Some help here about what format the version should be in would be nice.

Reply:
In light of @mvines's comments below, I gather the preferred resolution would be either to:

1. make this an optional flag that means "publish snapshots in an older format" as opposed to in a newer format which would be the default (and which would be in fact the same format os the older one most of the time), or

2. remove the CLI option entirely

I'd be inclined to go with #1 as it gives you a bit more flexibility in updates - i.e. rollout the binaries, allow time for node operators to update and then activate the new format; it decouples the binary change with the config change and allows for rollback without reverting the binary (*of course this is blockchain not banking so change control is rather different)

Of course with #2 the snapshot_version field in AccountsPackage is no longer needed and some of the edits can be taken out

Reply:
I'm game for #2.  Supporting multiple snapshot versions is a burden that I wouldn't want to carry for long, and I can't imagine caring about pushing a snapshot in the older format once the newer format is available.   We still have the luxury of being able to easily ask everybody to upgrade and get a result within 24/48 hours, which is pretty awesome.   One fine day we won't have that and then we'll need more upgrade machinery (like what you propose!)